modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/events/CommerceContextPreAction.java

```diff
@@ -47,7 +49,10 @@ public void run(
 			CommerceWebKeys.COMMERCE_CONTEXT, commerceContext);
 	}
 
-	@Reference
-	private CommerceContextFactory _commerceContextFactory;
+	@Reference(
```

Comment:
Why we need ReferencePolicyOption.GREEDY?

Reply:
So that the CommerceContextFactory with the highest service ranking will be used according to https://help.liferay.com/hc/en-us/articles/360022626112-Examining-an-OSGi-Service-to-Override-#step-3-gather-reference-configuration-details-if-reconfiguration-is-needed